### ISSUE TITLE
noUiSlider update event in Angular not updating view on slide but only when slider is released

### ISSUE THREAD

**Issue by HamzaNaeem1990** [leongersen/noUiSlider#980]
Hi, 

I am using the noUiSlider and not the ng2 noUiSlider in an angular app.I am trying to use noUiSlider update event to update a span element to show the values being set as i am sliding the handle but it is only display on view when the handle is released even though i am assigning the value in an update event. I have asked the following question on Stack overflow but no response yet. Can any one guide me in what i am doing wrong please? 

Question: https://stackoverflow.com/q/56170565/10681259

**Comment by leongersen** [comment]
I saw your stackoverflow question earlier, but I don't know enough about Angular to help you. Your slider code looks correct to me.

**Comment by HamzaNaeem1990** [comment]
What is the difference between using the update in a function like this:         
slider.on('update', function (this, values, handle) {
            this.selectedMinPrice = values[0];
            this.selectedMaxPrice = values[1];
        });

because this way does not update the view at all.. where as the way i have done it in my question it does update the view but only after release of the handle?

In both examples the console.log is updating as i am sliding the handle.

**Comment by leongersen** [comment]
When you use a `function` like that, `this` refers to the `noUiSlider` API, not the outside scope. In the arrow function in your original question, it is the other way around.

**Comment by HamzaNaeem1990** [comment]
Okay so using it in a function like this, Angular does not detect a change on the this.selectedMinPrice hence it does not update the view?

**Comment by leongersen** [comment]
`slider.on('update', function (this, values, handle) ` = `function`, so rebound scope.

`private sliderUpdate = (values: any[], handle) => {` = arrow function, so scope is maintained.

Try adding `console.log(this)` to see what is happening.

**Comment by HamzaNaeem1990** [comment]
So did console.log(this) and inside a function only the noUislider properties are available to m. Where as outside the function all the variables are available. So you are right inside a function it is out of scope. 

Thanks I am doing it the update the correct way but still only updating view on end....

**Comment by HamzaNaeem1990** [comment]
If i remove the `slider.on('end', this.sliderEnd);` event then my view will not update. Even though i am setting the two variables only inside the update event.

**Comment by leongersen** [comment]
Closing, as this issue seems to have gone stale. Feel free to comment.
